application/src/main/java/org/togetherjava/tjbot/commands/moderation/ModerationUtils.java

```diff
@@ -85,6 +85,41 @@ static boolean handleCanInteractWithTarget(@NotNull String actionVerb, @NotNull
         return true;
     }
 
+    /**
+     * Checks whether the given author and bot can interact with the given role. For example whether
+     * they have enough permissions to add or remove this role to users.
+     * <p>
+     * If not, it will handle the situation and respond to the user.
+     *
+     * @param bot the bot attempting to interact with the user
+     * @param author the author triggering the command
+     * @param role the role to interact with
+     * @param event the event used to respond to the user
+     * @return Whether the author and bot can interact with the role
+     */
+    @SuppressWarnings("BooleanMethodNameMustStartWithQuestion")
+    static boolean handleCanInteractWithRole(@NotNull Member bot, @NotNull Member author,
+            @NotNull Role role, @NotNull Interaction event) {
+        if (!author.canInteract(role)) {
+            event
+                .reply("The role %s is too powerful for you to interact with."
+                    .formatted(role.getAsMention()))
+                .setEphemeral(true)
+                .queue();
+            return false;
+        }
+
+        if (!bot.canInteract(role)) {
+            event
+                .reply("The role %s is too powerful for me to interact with."
+                    .formatted(role.getAsMention()))
+                .setEphemeral(true)
+                .queue();
+            return false;
+        }
+        return true;
+    }
+
```

Comment:
Can you not make this into an embed makes it look neater.

Reply:
It is not an embed. Did you fuck up your english and meant to say we should use an embed instead?

Would also have to be aligned with the other moderation commands where we are not using embeds for error messages. And with all other commands as well.

Id say, open a GH issue to discuss this for all commands (tex, tag, ... they all do it like that).